### PR TITLE
chore(custom-resources): update python runtime for custom resources

### PR DESCRIPTION
**Pull request by GavinZZ**
### Issue # (if applicable)

Closes https://github.com/aws/aws-cdk/issues/31245

### Reason for this change

Support latest python runtime that's available everywhere

### Description of changes

Update custom resource handler runtime to python 3.11

### Description of how you validated changes

There were initially 40+ failed integration tests due to this change, all these tests are re-ran by deploying and verifying (no `dry-run` flag used). Now all existing tests are passing.  

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by moelasmar — APPROVED** [review]

File: packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/config.ts [hunk 1]
```diff
@@ -10,16 +10,6 @@ export enum Runtime {
...
-  /**
-   * The Python 3.9 runtime
-   */
-  PYTHON_3_9 = 'python3.9',
-
-  /**
-   * The Python 3.10 runtime
-   */
-  PYTHON_3_10 = 'python3.10',
-
```
Comment (moelasmar) [inline packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/config.ts:22]: is it mandatory to remove these values ... this is an exposed Enum, and may be some customers are using these values .. do you think it is not a correct understanding

Comment (GavinZZ) [reply packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/config.ts:22]: I think you raised a good point. Although there is a better place to use the ENUM from ([here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-lambda/lib/runtime.ts#L53)), Customers might be using these runtime enum for some unknown reason. Let me add it back.

**Comment by GavinZZ** [comment]
@mergify update

**Comment by GavinZZ** [comment]
@mergify update

**Comment by GavinZZ** [comment]
@mergify update

**Review by moelasmar — APPROVED** [review]

**Comment by paulhcsun** [comment]
@mergifyio update

**Comment by GavinZZ** [comment]
@mergify refresh

**Pull request merged**